File: Quick_Sort.py
import numpy
def sort_as(mass, first, last):
    f=first
    l=last
    mid=mass[int((f+l)/2)]
    while(mass[f]<mid):
        f+=1
    while(mass[l]>mid):
        l-=1
    if(f<=l):
        temp=mass[f]
        mass[f]=mass[l]
        mass[l]=temp
        f+=1
        l-=1
    if(first<l):
        sort_as(mass,first,l)
    if(f<last):
        sort_as(mass,f,last)
    return mass
def sort_des(mass, first, last):
    f=first
    l=last
    mid=mass[int((f+l)/2)]
    while(mass[f]>mid):
        f+=1
    while(mass[l]<mid):
        l-=1
    if(f<=l):
        temp=mass[f]
        mass[f]=mass[l]
        mass[l]=temp
        f+=1
        l-=1
    if(first<l):
        sort_des(mass,first,l)
    if(f<last):
        sort_des(mass,f,last)
    return mass
n = int(input("Enter size of array"))
mass = numpy.random.randint(1000, size=n)
for i in range(n):
    print("", mass[i]," ",end="")
print("")
a = int(input("Choose how to sort: ascending or descending(enter 1 or 2 respectively)."))
if(a==1):
    sort_as(mass, 0, n-1)
if(a==2):
    sort_des(mass, 0, n-1)
print("The array is sorted: ",end="")
for i in range(n):
    print("", mass[i]," ",end="")

File: Quick_sort.java
import java.util.Scanner;

public class Quick_sort {
	static void sort_as(int[] vec, int first, int last)
	{
		int temp = 0, mid = 0;
		int f = first, l = last;
		mid = vec[(l + f) / 2];
		while (vec[f] < mid)
		{
			f++;
		}
		while (vec[l] > mid)
		{
			l--;
		}
		if (f <= l)
		{
			temp = vec[f];
			vec[f] = vec[l];
			vec[l] = temp;
			f++;
			l--;
		}
		if (first < l)
			sort_as(vec, first, l);
		if (f < last)
			sort_as(vec, f, last);
	}
	static void sort_des(int[] vec, int first, int last)
	{
		int temp = 0, mid = 0;
		int f = first, l = last;
		mid = vec[(l + f) / 2];
		while (vec[f] > mid)
		{
			f++;
		}
		while (vec[l] < mid)
		{
			l--;
		}
		if (f <= l)
		{
			temp = vec[f];
			vec[f] = vec[l];
			vec[l] = temp;
			f++;
			l--;
		}
		if (first < l)
			sort_des(vec, first, l);
		if (f < last)
			sort_des(vec, f, last);
	}


	public static void main(String[] args) {
		int n = 0;
		Scanner in = new Scanner(System.in);
		System.out.println("Enter size of array");
		n=in.nextInt();
		int[] mass = new int[n];
		for(int i=0;i<n;i++)
		{
			mass[i] = (int) ( Math.random() * 1000);
			System.out.print(mass[i]+" ");
		}
		short a=0;
		System.out.println();
		System.out.println("Choose how to sort: ascending or descending(enter 1 or 2 respectively).");
		a=in.nextShort();
		switch(a)
		{
			case 1:sort_as(mass, 0, n-1); break;
			case 2:sort_des(mass, 0, n-1); break;
			default: break;
		}
		System.out.print("The array is sorted: ");
		for(int i=0;i<n;i++)
		{
			System.out.print(mass[i]+" ");
		}	
		

	}

}
